aleph0/index.ts

```diff
@@ -64,19 +68,43 @@ async function generate(userText: string) {
   consola.log(changes);
 
   // 3. For each file in the changes array, ask GPT 4 for the new file and create/modify it.
+  const RELATIVE_DIR = "../examples/next";
   consola.info(
-    `Step 3 - for each file in the changes array, ask GPT 4 for the new file and create/modify it.`
+    `\nStep 3 - for each file in the changes array, ask GPT 4 for the new file and create/modify it.`
   );
   for (const change of changes) {
-    consola.log(`Change: ${JSON.stringify(change, null, 2)}`);
+    consola.log(`Change operation: ${JSON.stringify(change, null, 2)}`);
+
+    const sourceFilePath = path.join(RELATIVE_DIR, change.sourcePath);
+    const sourceFile = Bun.file(sourceFilePath);
     const snippet = readFile(change.snippetPath);
-    const fileContents = await ai(
-      await generateFile({ snippet, userText, specificKnowledge }),
-      undefined,
-      "gpt-4"
-    );
+
+    let fileContents;
+    if (await sourceFile.exists()) {
+      consola.info("Updating existing file");
+      const currentFileContents = await sourceFile.text();
+      fileContents = await ai(
+        await updateFilePrompt({
+          snippet,
+          userText,
+          specificKnowledge,
+          fileContents: currentFileContents,
```

Comment:
At some point we'll have to handle going over max tokens. But I think we're okay now with 8k limit. Which I believe is around 6k words.